parcels/compilation/codeinterface.py

```diff
@@ -0,0 +1,401 @@
+            self.src_file = "%s.c" % (os.path.join(src_dir, src_pathfile), )
+        self.lib_file = "%s.%s" % (os.path.join(get_cache_dir(), lib_path), libext)
+        self.log_file = "%s.log" % (os.path.join(get_cache_dir(), lib_path), )
+        if os.path.exists(self.lib_file):
+            self.compiled = True
+
+        self.compiler = compiler
+        self.compiled = False
+        self.loaded = False
+        self.libc = None
+        self.register_count = 0
+        self.close_cb = []
+
+    def __del__(self):
+        """
+        C-Interface - Destructor
+        """
+        self.unload_library()
+        self.cleanup_files()
+
+    def is_compiled(self):
+        """
+        :returns if this C-library is compiled or not
+        """
+        return self.compiled
+
+    def is_loaded(self):
+        """
+        :returns if this C-library is loaded or not
+        """
+        return self.loaded
+
+    def is_registered(self):
+        """
+        :returns if any external object has been registered with this library or not
+        """
+        return self.register_count > 0
+
+    def get_library_path(self):
+        """
+        :returns the path to the library-file, incl. the library filename itself
+        """
+        return self.lib_file
+
+    def get_library_dir(self):
+        """
+        :returns the path to the directory where the library-file is located
+        """
+        return os.path.dirname(self.lib_file)
+
+    def get_library_basename(self):
+        """
+        :returns the name of the library filename, without the parent directory or the file extension
+        """
+        return self.basename
+
+    def get_library_filename(self):
+        """
+        :returns the library filename, without the parent directory or the file extension; commonly similar
+                 to 'get_library_basename()', but can include the 'lib'-suffix on UNIX platforms
+        """
+        return os.path.basename(self.lib_file)
+
+    def get_library_extension(self):
+        """
+        :returns the file extension of the library
+        """
+        return os.path.splitext(self.lib_file)[1]
+
+    def compile_library(self):
+        """ Writes kernel code to file and compiles it."""
+        if not self.compiled:
+            src_file = self.src_file if not isinstance(self.compiler, CCompiler_MS) or (isinstance(self.src_file, list) and not isinstance(self.src_file, str)) else [self.src_file, ]
+            # logger.info("InterfaceC::compile_library() - 'src_file' = {}".format(src_file))
+            compiled_fpath = self.compiler.compile(src=src_file, obj=self.lib_file, log=self.log_file)
+            # logger.info("Compiled %s ==> %s" % (self.basename, self.lib_file))
+            logger.info("Compiled %s ==> %s" % (self.basename, compiled_fpath))
+            assert os.path.exists(compiled_fpath)
+            # self._cleanup_files = finalize(self, package_globals.cleanup_remove_files, self.lib_file, self.log_file)
```

Comment:
Stray debug comments?
```suggestion
            compiled_fpath = self.compiler.compile(src=src_file, obj=self.lib_file, log=self.log_file)
            logger.info("Compiled %s ==> %s" % (self.basename, compiled_fpath))
            assert os.path.exists(compiled_fpath)
```